numba/datamodel/models.py

```diff
@@ -1105,6 +1105,41 @@ def handle_bound_function(dmm, ty):
     return dmm[ty.this]
 
 
+@register_default(types.NumpyNdIterType)
+class NdIter(StructModel):
+    def __init__(self, dmm, fe_type):
```

Comment:
I don't think it needs a docstring. It's the same constructor signature as for all models, and the class's purpose itself is self-explanatory (it's the datamodel for `NumpyNdIterType` instances, as the decorator indicates).